lib/mllp/sender.ex

```diff
@@ -343,6 +314,48 @@ defmodule MLLP.Sender do
     stop_connection(state, reason, "process terminated")
   end
 
+  @should_reconnect_errors [
+    :closed,
+    :enotconn,
+    :econnreset,
+    :enetreset,
+    :econnaborted,
+    :econnrefused
+  ]
+  defp handle_send_error(state, error) when error in @should_reconnect_errors do
+    telemetry(
+      :status,
+      %{status: :disconnected, error: error, context: "send message failure"},
+      state
+    )
+
+    new_state = maintain_reconnect_timer(state)
+    reply = {:error, %{type: :send_failure, reason: error}}
+    {:reply, reply, new_state}
+  end
+
+  defp handle_send_error(state, error) do
+    # Add telemetry call here
```

Comment:
I think this should be handled in another ticket.

Reply:
If this ever happens and we are not retrying to connect, a caller can call send message again on failure and it will most likely fail again as socket never probably recovered from previous error. Would we leave caller in a state they can't come out from without killing this process. Having this state would be bad.

Reply:
Should we close the socket in this situation or we let the process crash?

Reply:
I think I said to myself, let the caller decide on this one. This  is not a case we can clearly handle, so it may be best letting the caller decide. Thoughts?